ranger/blackhole_vm_metadata_server.sh

```diff
@@ -0,0 +1,17 @@
+set -xe
+
+# Blackhole the VM metadata server except for root.
+iptables -A OUTPUT -d 169.254.169.254 -p tcp -m owner --uid-owner root -j ACCEPT
+iptables -A OUTPUT -d 169.254.169.254 -p tcp -j REJECT
+
+# Install iptables-persistent to preserve the setting after reboot.
+# This section only works on Debian and Ubuntu (Debian based systems).
+echo iptables-persistent iptables-persistent/autosave_v4 boolean true | debconf-set-selections
+echo iptables-persistent iptables-persistent/autosave_v6 boolean true | debconf-set-selections
+apt-get update
+apt-get -y install iptables-persistent
```

Comment:
Aren't this already installed in all Dataproc images?

Reply:
We have iptabes save, but using that solely requires users to use iptables restore to restore the setting. iptables persistent basically do that for you when it is loaded into the OS on boot.

Reply:
Interesting, probably we need to use it in our own images then too...